docs/building-blocks/extending-templates/conditional-linking.md

```diff
@@ -0,0 +1,173 @@
+---
+title: Extending Azure Resource Manager Template functionality - Conditional Template Linking
+description: Describes how to extend the functionality of Azure Resource Manager templates to implement conditional template linking
+author: petertay
+ms.service: guidance
+ms.topic: article
+ms.date: 05/03/2017
+ms.author: pnp
+
+---
+# Azure Resource Manager templates - conditional template linking
+
+Azure Resource Manager templates support linking from one template file to another using the `Microsoft.Resources/deployments` resource type. You can specify resources in a seperate template and link to them using the `templateLink` property. This is great for managing templates and making deployment more easily configurable.
+
+However, there is a limitation in that templates can only be statically linked when you create the main template. If you want to have two sets of similar resources, specify them in separate templates, and deploy one or the other under certain conditions, you must determine the condition in advance and manually change the path to the linked template before deployment begins.
+
+For example, for some deployments you may require a VNet with an external domain name server (DNS) but for others require Azure DNS. To do this, you'd have to specify each VNet in a separate template file and manually edit the main template to link to one or the other based on which you want to deploy.
+
+The problem is that resource manager does not include functionality to conditionally link templates using parameters. But there is a way to implement this using existing resource manager functions.
+
+In the following template, a variable includes an object with two or more properties. The two or more properties specify the `uri` of a template. Then, a `Microsoft.Resources/deployments` resource type is specified with a `templateLink` property whose `uri` is the variable object indexed by a parameter value. Another template specifies a value for the parameter, and during deployment this parameter value changes the index and this in turn changes the `uri`. 
```

Comment:
I had to read this paragraph a few times - I think it would help to use the specific names from the example. The `conditions` variable contains an object with two properties, `yes` and `no`. Each of these properties specifies the URI of a template.  The `condition` parameter, which is specified during deployment, is used to select one of the two properties.  (something like this, anyway)